ouroboros-consensus-shelley-test/src/Test/ThreadNet/TxGen/Shelley.hs

```diff
@@ -62,8 +61,6 @@ instance HashAlgorithm h => TxGen (ShelleyBlock (TPraos (MockCrypto h)) (MockShe
       --
       -- When fixed, remove the line below to re-enable the transaction
       -- generator.
-      | otherwise               = pure []
-
```

Comment:
This is a logic change, isn't it?

Reply:
Furthermore, if the code above the deleted code  did not cover **all** cases, there would have been a compiler warning and there wasn't.

I also find it useful to convert `-Wall` warnings to errors using `-Werror` . I trust GHC far more than I trust myself.

Reply:
The problem is not that not all cases all covered, but rather that there is a logic change 

The code in question looked like this:

```haskell
  testGenTxs _coreNodeId _numCoreNodes curSlotNo cfg extra lst
      | stgeStartAt > curSlotNo = pure []

      -- TODO Temporarily disable the transaction generator until we fix the
      -- failing assertion in TxSubmission.Inbound, see #2680.
      --
      -- When fixed, remove the line below to re-enable the transaction
      -- generator.
      | otherwise               = pure []

      | otherwise               = do
      n <- choose (0, 20)
      go [] n $ applyChainTick lcfg curSlotNo lst
```

The first `| otherwise` clause was intentionally added in order to work around an existing bug (#2680). Now the pattern match redundancy checker got smarter in GHC 9.2 compared to 8.10, and detects that the second `| otherwise` clause is unreachable (which is intentional here!).

To fix this, you can use a more "complicated" expression that evaluates to `True` instead of `otherwise`, e.g. `() == ()` (or even the new [`considerAccessible`](https://hackage.haskell.org/package/base-4.17.0.0/docs/GHC-Exts.html#v:considerAccessible) that has exactly this purpose).

Does that clear it up?

---

> I also find it useful to convert `-Wall` warnings to errors using `-Werror`.

Agreed, we do that on Hydra, but this PR does not add 9.2 to Hydra CI (see https://github.com/input-output-hk/ouroboros-network/pull/4144#issuecomment-1322006034).

Reply:
The current way the last case is disabled is confusing to people not a part of this project in that it current generates a compiler warning that someone not familiar with the code thinks they should remove.

Reply:
I have come up with an alternative that both maintains the logic and satisfies GHC.

Reply:
Thanks, using an `if True` is another way to "outsmart" even the improved GHC 9.2 pattern match redundancy checker :smile: